nixos/modules/programs/nix-ld.nix

```
@@ -2,15 +2,14 @@
 let
   cfg = config.programs.nix-ld;
 
-  # TODO make glibc here configurable?
-  nix-ld-so = pkgs.runCommand "ld.so" {} ''
-    ln -s "$(cat '${pkgs.stdenv.cc}/nix-support/dynamic-linker')" $out
-  '';
-
   nix-ld-libraries = pkgs.buildEnv {
     name = "lb-library-path";
     pathsToLink = [ "/lib" ];
     paths = map lib.getLib cfg.libraries;
+    # TODO make glibc here configurable?
+    postBuild = ''
+      ln -s "$(cat '${pkgs.stdenv.cc}/nix-support/dynamic-linker')" $out/share/nix-ld/lib/ld.so
```

Seems a bit leaner to rely on this now…

```suggestion
      ln -s "${pkgs.stdenv.cc.bintools.dynamicLinker}" $out/share/nix-ld/lib/ld.so
```

---

That's a new one, I wasn't aware of.